backend/utils/cache-posts.js

```diff
@@ -0,0 +1,32 @@
+import { redis } from '../config/redis.js';
+import { REDIS_PREFIX } from './constants.js';
+
+// Helper function to check if Redis is available
+function isRedisEnabled() {
+  return redis !== null;
+}
+
+export async function getKeyFromCache(key) {
```

Comment:
Better to use `retrieveDataFromCache`, and `storeDataInCache`

Reply:
[OPTIONAL]: You can create a single service which will take care of redis connection, redis client and supporting methods/features.